### PR TITLE
LPS-184523 LPS-188356 User ERC instead of ID Blueprints

### PR DESCRIPTION
**Pull request by gustavolimav**
**Story**
https://liferay.atlassian.net/browse/LPS-184523

**Subtasks**
https://liferay.atlassian.net/browse/LPS-186826
https://liferay.atlassian.net/browse/LPS-186827
https://liferay.atlassian.net/browse/LPS-186828

Those first 3 were already forwarded but Brian Chan asked for some changes -> https://github.com/brianchandotcom/liferay-portal/pull/136614#issuecomment-1598830245

changes made [here](https://github.com/liferay-search/liferay-portal/pull/971/commits/26ab6c8f14ba3c421d88bda3e382187211ab6087)

https://liferay.atlassian.net/browse/LPS-187222
https://liferay.atlassian.net/browse/LPS-187223
https://liferay.atlassian.net/browse/LPS-188093
https://liferay.atlassian.net/browse/LPS-188356

Before Upgrade suggestionsContributorConfigurations:

> {"contributorName":"basic","displayGroupName":"suggestions","size":5},{"attributes":{"characterThreshold":"0","fields":[],"includeAssetSearchSummary":true,"includeAssetURL":true,"sxpBlueprintId":33881},"contributorName":"sxpBlueprint","displayGroupName":"Pages","size":"3"}

After Upgrade:

> {"contributorName":"basic","displayGroupName":"suggestions","size":5},{"attributes":{"characterThreshold":"0","fields":[],"includeAssetSearchSummary":true,"includeAssetURL":true,"sxpBlueprintExternalReferenceCode":"e0ea8462-8c7b-d361-f61b-0aa75fa428d9"},"contributorName":"sxpBlueprint","displayGroupName":"Pages","size":"3"}

### CODE REVIEW

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/search_bar_configuration_suggestions/contributor_input_set_item/inputs/SXPBlueprintSelectorInput.js [hunk 1]
```diff
@@ -94,14 +95,14 @@ function SXPBlueprintSelectorInput({
...
-								? JSON.stringify(sxpBlueprintId)
+								? sxpBlueprintExternalReferenceCode
 								: data.title,
 					});
 				})
 				.catch(() => {
 					setSXPBlueprint({
 						loading: false,
-						title: JSON.stringify(sxpBlueprintId),
+						title: sxpBlueprintExternalReferenceCode,
```
Comment (oliv-yu) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/search_bar_configuration_suggestions/contributor_input_set_item/inputs/SXPBlueprintSelectorInput.js:105]: My SF does this:

	selectedExternalReferenceCode={
		sxpBlueprintExternalReferenceCode || ''
	}

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:relevant

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v2_0_0/SXPBlueprintUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+					"where portletId like ", portletIdQuoted))) {
+
+			ResultSet resultSet = preparedStatement1.executeQuery();
+
+			_upgradeSXPBlueprintOptionsPortlets(resultSet);
+		}
+	}
+
+	private void _upgradeSmallValueAndName(
+			String portletPreferencesIdQuoted, ResultSet resultSet2)
```
Comment (peerkar) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v2_0_0/SXPBlueprintUpgradeProcess.java:176]: Just resultSet instead of resultSet2. Please check all the numbered variables in this class as they seem to start from random numbers. 

I assume this happened because of copy & paste, when refactoring the code and splitting the logic.

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/search_bar_configuration_suggestions/contributor_input_set_item/inputs/SXPBlueprintSelectorInput.js [hunk 3]
```diff
@@ -115,7 +115,9 @@ function SXPBlueprintSelectorInput({
 					observer={observer}
 					onClose={onClose}
 					onSubmit={_handleSubmit}
-					selectedExternalReferenceCode={sxpBlueprintExternalReferenceCode || ''}
+					selectedExternalReferenceCode={
```
Comment (peerkar) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/search_bar_configuration_suggestions/contributor_input_set_item/inputs/SXPBlueprintSelectorInput.js:118]: Brian may not like SFs to be separate commits, but this is not much so could be fine

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/suggestions/spi/SXPBlueprintSuggestionsContributor.java [hunk 4]
```diff
@@ -97,11 +97,13 @@ public SuggestionsContributorResults getSuggestionsContributorResults(
 			(Map<String, Object>)
 				suggestionsContributorConfiguration.getAttributes();
 
-		if ((attributes == null) || !attributes.containsKey("sxpBlueprintId")) {
+		if ((attributes == null) ||
+			!attributes.containsKey("sxpBlueprintExternalReferenceCode")) {
```
Comment (peerkar) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/suggestions/spi/SXPBlueprintSuggestionsContributor.java:101]: I think this contributor should keep the possibility to rely also on ID because afaik there already are customers using the REST endpoint directly from their widget templates. Upgrade processes won't cover / help them. 

Otherwise this would be a breaking change. cc: @lipusz

**Comment by peerkar** [comment]
@gustavolimav just a few small things to check there and it looks good to go from my side.
I didn't try out the upgrade processes, but I assume you did :)

**Comment by xbrianlee** [comment]
ci:forward:force
